orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -114,6 +115,9 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
         transformers.emplace_back(onnxruntime::make_unique<TransformerLayerRecompute>(
             config.number_recompute_layers, compatible_eps));
       }
+      if (config.propagate_cast_ops_level > 0) {
+        transformers.emplace_back(onnxruntime::make_unique<PropagateCastOps>(config.propagate_cast_ops_level, config.propagate_cast_ops_allow, compatible_eps));
```

Comment:
BTW, this Transformer should only be enabled for runs with CUDA EP. 
CPU doesn't support fp16 computation

Reply:
passing CUDA ep instead of compatible_eps